components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -214,7 +214,7 @@ func (agent *Smith) Start(ctx context.Context, callback func(InfringingWorkspace
 	agent.metrics.RegisterClassificationQueues(cli, clo)
 
 	workspaces := make(map[int]*common.Workspace)
-	wssWriteMutex := &sync.Mutex{}
+	wsMutex := &sync.Mutex{}
```

Comment:
Hey there, you may also be able to try this `sync.RWMutex`, it should allow for a single writer and one or more readers.

@princerachit @utam0k